### PR TITLE
Investigate fixes

### LINKED ISSUE #2791: Malicious Affliction copy isn't optional
When casting Malicious Affliction, the morbid condition automatically copies it, but this should be a "may" effect.

### PR DESCRIPTION
**Pull request by tool4ever**
- Fix one opponent declining stopping the investigating on _Will the Wise_ completely

- Closes #2791

### CODE REVIEW

**Review by Agetian — DISMISSED** [review]

File: forge-gui/res/cardsfolder/h/haakon_stromgald_scourge.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ Types:Legendary Creature Zombie Knight
 PT:3/3
 S:Mode$ Continuous | Affected$ Card.Self | MayPlay$ True | AffectedZone$ Graveyard | EffectZone$ Graveyard
 S:Mode$ CantBeCast | ValidCard$ Card.Self | Origin$ Exile,Hand,Library,Command | EffectZone$ Graveyard,Hand,Library,Command,Stack | Description$ You may cast CARDNAME from your graveyard, but not from anywhere else.
-S:Mode$ Continuous | Affected$ Knight.YouCtrl | MayPlay$ True | EffectZone$ Battlefield | AffectedZone$ Graveyard | Description$ As long as CARDNAME is on the battlefield, you may play Knight cards from your graveyard.
+S:Mode$ Continuous | Affected$ Knight.YouCtrl | ValidAfterStack$ Spell.Knight | MayPlay$ True | EffectZone$ Battlefield | AffectedZone$ Graveyard | Description$ As long as CARDNAME is on the battlefield, you may cast Knight spells from your graveyard.
 T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigLose | TriggerDescription$ When CARDNAME dies, you lose 2 life.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/h/haakon_stromgald_scourge.txt:8]: bah as long as we're here might as well NICKNAME

File: forge-gui/res/cardsfolder/h/haakon_stromgald_scourge.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ Types:Legendary Creature Zombie Knight
 PT:3/3
 S:Mode$ Continuous | Affected$ Card.Self | MayPlay$ True | AffectedZone$ Graveyard | EffectZone$ Graveyard
 S:Mode$ CantBeCast | ValidCard$ Card.Self | Origin$ Exile,Hand,Library,Command | EffectZone$ Graveyard,Hand,Library,Command,Stack | Description$ You may cast CARDNAME from your graveyard, but not from anywhere else.
-S:Mode$ Continuous | Affected$ Knight.YouCtrl | MayPlay$ True | EffectZone$ Battlefield | AffectedZone$ Graveyard | Description$ As long as CARDNAME is on the battlefield, you may play Knight cards from your graveyard.
+S:Mode$ Continuous | Affected$ Knight.YouCtrl | ValidAfterStack$ Spell.Knight | MayPlay$ True | EffectZone$ Battlefield | AffectedZone$ Graveyard | Description$ As long as CARDNAME is on the battlefield, you may cast Knight spells from your graveyard.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/h/haakon_stromgald_scourge.txt:7]: NICKNAME here too

File: forge-gui/res/cardsfolder/t/tempting_contract.txt [hunk 3]
```diff
@@ -2,10 +2,10 @@ Name:Tempting Contract
 ManaCost:4
 Types:Artifact
 T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ DBRepeat | SubAbility$ DBToken | TriggerDescription$ At the beginning of your upkeep, each opponent may create a Treasure token. For each opponent who does, you create a Treasure token.
-SVar:DBRepeat:DB$ RepeatEach | RepeatSubAbility$ DBOppToken | RepeatPlayers$ Player.Opponent | SubAbility$ DBToken | RepeatOptionalForEachPlayer$ True | RepeatOptionalMessage$ Do you want to create a Treasure token?
-SVar:DBOppToken:DB$ Token | TokenScript$ c_a_treasure_sac | TokenOwner$ Player.IsRemembered | RememberTokens$ True
+SVar:DBRepeat:DB$ RepeatEach | RepeatSubAbility$ DBOppToken | RepeatPlayers$ Player.Opponent | RepeatOptionalForEachPlayer$ True | RememberDeciders$ True | RepeatOptionalMessage$ Do you want to create a Treasure token? | ChangeZoneTable$ True | SubAbility$ DBToken
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/t/tempting_contract.txt:5]: `RepeatPlayers$ Opponent` better, right?

File: forge-gui/res/cardsfolder/t/tempting_contract.txt [hunk 4]
```diff
@@ -2,10 +2,10 @@ Name:Tempting Contract
...
+SVar:DBOppToken:DB$ Token | TokenScript$ c_a_treasure_sac | TokenOwner$ Player.IsRemembered
 SVar:DBToken:DB$ Token | TokenAmount$ X | TokenScript$ c_a_treasure_sac | SubAbility$ DBCleanup
 SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
-SVar:X:Remembered$Amount
+SVar:X:Count$RememberedSize
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/t/tempting_contract.txt:9]: So nice that we can get away with this kind of thing now even when using RepeatEach!

**Review by Northmoc — APPROVED** [review]
👍

**Pull request merged**